kornia/color/colormap.py

```diff
@@ -0,0 +1,183 @@
+from abc import ABC
+from typing import List, NamedTuple, Union
+
+import torch
+from torch.nn.functional import interpolate
+
+from kornia.core import Module, Tensor, tensor
+from kornia.testing import KORNIA_CHECK_IS_GRAY
+
+
+class RGBColor(NamedTuple):
+    r"""RGB color representation."""
+    red: Union[int, float, Tensor]
+    green: Union[int, float, Tensor]
+    blue: Union[int, float, Tensor]
+
+
+def _lst_color_to_tensor(colors: List[RGBColor], dtype: torch.dtype, device: torch.device) -> Tensor:
+    return tensor([list(color) for color in colors], dtype=dtype, device=device).T
+
+
+class ColorMap(ABC):
+    r"""Abstract class to represents a color map."""
+    colors: List[RGBColor]
+
+    def __init__(self, base_colormap: List[RGBColor], n: int, device: torch.device, dtype: torch.dtype):
+        self._dtype = dtype
+        self._device = device
+
+        _lut = interpolate(
+            _lst_color_to_tensor(base_colormap, dtype=self._dtype, device=self._device)[None, :], size=n, mode='linear'
+        )
+        self.colors = [RGBColor(red=_lut[0, 0, x], green=_lut[0, 1, x], blue=_lut[0, 2, x]) for x in range(n)]
+
+    def __len__(self) -> int:
+        return len(self.colors)
+
+    def get_colors(self) -> Tensor:
+        return tensor([list(color) for color in self.colors], device=self._device, dtype=self._dtype).squeeze(-1)
+
+
+# TODO: Add more colormaps
+_BASE_AUTUMN = [
+    RGBColor(red=1, green=0, blue=0),
```

Comment:
i like here being explicit but a bit worried about contributing with more style. Maybe it's easier to just define as list or tuple with proper docs and later construct the `RGBColor` object ?